tests/unit_tests/indicators/rust/test_rma_pyo3.py

```diff
@@ -0,0 +1,157 @@
+
+
+def test_name_returns_expected_string(rma: WilderMovingAverage):
+    # Arrange, Act, Assert
+    assert rma.name == "WilderMovingAverage"
+
+
+def test_str_repr_returns_expected_string(rma: WilderMovingAverage):
+    # Arrange, Act, Assert
+    assert str(rma) == "WilderMovingAverage(10)"
+    assert repr(rma) == "WilderMovingAverage(10)"
+
+
+def test_period_returns_expected_value(rma: WilderMovingAverage):
+    # Arrange, Act, Assert
+    assert rma.period == 10
+
+
+def test_multiplier_returns_expected_value(rma: WilderMovingAverage):
+    # Arrange, Act, Assert
+    assert rma.alpha == 0.1
+
+
+def test_initialized_without_inputs_returns_false(rma: WilderMovingAverage):
+    # Arrange, Act, Assert
+    assert rma.initialized is False
+
+
+def test_initialized_with_required_inputs_returns_true(rma: WilderMovingAverage):
+    # Arrange
+    rma.update_raw(1.00000)
+    rma.update_raw(2.00000)
+    rma.update_raw(3.00000)
+    rma.update_raw(4.00000)
+    rma.update_raw(5.00000)
+    rma.update_raw(6.00000)
+    rma.update_raw(7.00000)
+    rma.update_raw(8.00000)
+    rma.update_raw(9.00000)
+    rma.update_raw(10.00000)
+
+    # Act
+
+    # Assert
+    assert rma.initialized is True
+
+
+def test_handle_quote_tick_updates_indicator():
+    # Arrange
+    indicator = WilderMovingAverage(10, PriceType.MID)
+
+    tick = TestDataProviderPyo3.quote_tick()
+
+    # Act
+    indicator.handle_quote_tick(tick)
+
+    # Assert
+    assert indicator.has_inputs
+    assert indicator.value == 1987.5
+
+
+def test_handle_trade_tick_updates_indicator(rma: WilderMovingAverage):
+    # Arrange
+
+    tick = TestDataProviderPyo3.trade_tick()
+
+    # Act
+    rma.handle_trade_tick(tick)
+
+    # Assert
+    assert rma.has_inputs
+    assert rma.value == 1987.0
+
+
+def test_handle_bar_updates_indicator(rma: WilderMovingAverage):
+    # Arrange
+    bar = TestDataProviderPyo3.bar_5decimal()
+
+    # Act
+    rma.handle_bar(bar)
+
+    # Assert
+    assert rma.has_inputs
+    assert rma.value == 1.00003
+
+
+def test_value_with_one_input_returns_expected_value(rma: WilderMovingAverage):
+    # Arrange
+    rma.update_raw(1.00000)
+
+    # Act, Assert
+    assert rma.value == 1.0
+
+
+def test_value_with_three_inputs_returns_expected_value(rma: WilderMovingAverage):
+    # Arrange
+    rma.update_raw(1.00000)
+    rma.update_raw(2.00000)
+    rma.update_raw(3.00000)
+
+    # Act, Assert
+    assert rma.value == 1.29
+
+
+def test_value_with_ten_inputs_returns_expected_value(rma: WilderMovingAverage):
+    # Arrange
+    rma.update_raw(1.0)
+    rma.update_raw(2.0)
+    rma.update_raw(3.0)
+    rma.update_raw(4.0)
+    rma.update_raw(5.0)
+    rma.update_raw(6.0)
+    rma.update_raw(7.0)
+    rma.update_raw(8.0)
+    rma.update_raw(9.0)
+    rma.update_raw(10.0)
+
+    # Act, Assert
+    assert rma.value == 4.486784401
+
+
+def test_reset_successfully_returns_indicator_to_fresh_state(rma: WilderMovingAverage):
+    # Arrange
+    for _i in range(1000):
```

Comment:
You dont have to put `1000`. Want these tests to be fast and easy as possible. Just update few time, reset and tests it

Reply:
Reduced it to 10 iterations